### PR TITLE
Updated `.gitignore`

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

- ignore `rector.php` config file for now
- ignore files prefixed with test
- `qodana.yaml` from phpstorm
- `.phpunit.result.cache`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -80,3 +81,9 @@ phpstan*.neon
 # dev scripts loaded via composer
 /shell/update-copyright.php
 /shell/translations.php
+
+# test files
+**test*.php
```
Comment (kiatng) [inline .gitignore:86]: Sorry for my poor knowledge, but what does it match?

Comment (Flyingmana) [reply .gitignore:86]: if it only should match files prefixed with test, then I think there is a / missing and it should be `**/test*.php`

Comment (sreichel) [reply .gitignore]: @kiatng no default files, but in nearly every project hast some test files .;)

**Review by ADDISON74 — APPROVED** [review]

**Comment by fballiano** [comment]
I understand where this comes from but I wouldn't add those extensions to the ignore because on my final projects I may want to have those files in the repo

**Comment by sreichel** [comment]
I'd prefer "ignore all" first and only keep what belongs to core and have a customzed .gitignore in final project.

File: .gitignore (hunk 1 above)
Comment (kiatng) [reply .gitignore]: It's true, I have them, just not name with the word "test'.

**Pull request closed without merging**